apps/web/lib/hooks/usePublicPage.ts

```diff
@@ -2,7 +2,7 @@ import { usePathname } from "next/navigation";
 
 export default function usePublicPage() {
   const pathname = usePathname();
-  const isPublicPage = ["/[user]", "/booking", "/cancel", "/reschedule"].find((route) =>
+  const isPublicPage = ["/[user]", "/booking/", "/cancel", "/reschedule"].find((route) =>
```

Comment:
It was causing bookings/upcoming page to be considered public page and thus was removing Intercom Provider from there which was causing the DOM to be reconstructed for entire page.